rosbridge_server/scripts/rosbridge_websocket_client.py

```diff
@@ -64,46 +61,58 @@ def wsconnection_cb(self, conn):
         self.conn = conn.result()
         # TODO check result
         self.conn.on_message = self.message
-        self.send_message('{"op":"proxy"}')
+        self.conn.write_message('{"op":"proxy","user_auth":"True"}')
         self.keepalive = IOLoop.instance().add_timeout(
             timedelta(seconds=PING_TIMEOUT), self.dokeepalive)
 
     def message(self, _message):
         global user_auth
-        #print "Msg received: [%s]" % _message
         msg = json.loads(_message)
-        if msg['op'] == 'video':
-            try:
-                args = msg['args']
-                self.transfer = VideoTransfer("http://localhost:8080/stream",
-                                              args, self)
-            except Exception as e:
-                rospy.logerror("Could not connect to WebCam")
-                rospy.logerror(e)
-                self.send_message('{"op":"endVideo"}')
-        elif msg['op'] == "endVideo":
-            #TODO Resolve stop from client
-            self.transfer.endVideo()
-            pass
-        elif msg['op'] == 'auth':
+        session_id = msg['session_id']
+        protocol = None
+        if session_id != None:
+            protocol = protocols.get(session_id)
+        if protocol == None:
+            rospy.loginfo("New Protocol session %s" % session_id)
+            protocol = MyRosbridgeProtocol(session_id,self.conn,session_id)
+            protocols[session_id] = protocol
+            protocol.outgoing = protocol.send_message
+        if msg['op'] == 'auth':
             try:
                 # check the authorization information
-                if user_auth and not self.authenticated:
+                print "AUTH"
+                if user_auth:
+                    print "authenticating.."
                     auth_srv = rospy.ServiceProxy('/authenticate_user',
                                                   UserAuthentication)
                     resp = auth_srv(msg['user'], msg['pass'])
-                    self.authenticated = resp.authenticated
-                    if self.authenticated:
+                    if resp.authenticated:
                         rospy.loginfo("Client has authenticated")
-                        return
-                    # if we are here, no valid authentication was given
-                    rospy.logwarn("Client did not authenticate. Closing "
+                        self.conn.write_message(json.dumps({"op":"auth_client","session_id":msg['session_id'],"authentication":resp.authenticated }))
+                    else: 
+                        # if we are here, no valid authentication was given
+                        rospy.logwarn("Client did not authenticate. Closing "
                                   "connection.")
```

Comment:
How does disconnection of this client happen? Shouldn't this send a message back to the proxy server to let it know it should disconnect the client?